File: src/lc312/SolutionTest.java
package lc312;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class SolutionTest {

    @Test
    void testMaxCoins() {
        assertEquals(50, new Solution().maxCoins(new int[]{5, 2, 3}));
    }
}

File: src/lc44/SolutionTest.kt
package lc44

import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*

internal class SolutionTest {

    @Test
    fun isMatch() {
        assertTrue(Solution().isMatch("abefcdgiescdfimde", "ab*cd?i*de"))
    }
}